model/jpa/src/main/java/org/keycloak/models/jpa/JpaRealmProvider.java

```diff
@@ -547,22 +543,24 @@ public GroupModel createGroup(RealmModel realm, String id, String name, GroupMod
         GroupEntity groupEntity = new GroupEntity();
         groupEntity.setId(id);
         groupEntity.setName(name);
-        RealmEntity realmEntity = em.getReference(RealmEntity.class, realm.getId());
-        groupEntity.setRealm(realmEntity.getId());
+        groupEntity.setRealm(realm.getId());
         groupEntity.setParentId(toParent == null? GroupEntity.TOP_PARENT_ID : toParent.getId());
         em.persist(groupEntity);
         em.flush();
-        realmEntity.getGroups().add(groupEntity);
 
-        GroupAdapter adapter = new GroupAdapter(realm, em, groupEntity);
-        return adapter;
+        return new GroupAdapter(realm, em, groupEntity);
     }
 
     @Override
     public void addTopLevelGroup(RealmModel realm, GroupModel subGroup) {
         subGroup.setParent(null);
     }
 
+    @Override
+    public void preRemove(RealmModel realm, RoleModel role) {
+        em.createNamedQuery("deleteGroupRoleMappingsByRole").setParameter("roleId", role.getId()).executeUpdate();
```

Comment:
I believe this method will soon become overused from other parts as well (e.g. users). Could you clearly mark by comments that this particular command is there due to GroupProvider? This will become useful once we work on https://issues.redhat.com/browse/KEYCLOAK-14750.

Reply:
I meant really only to mark the lines within the method like 
```suggestion
        // GroupProvider interface method implementation starts here
        em.createNamedQuery("deleteGroupRoleMappingsByRole").setParameter("roleId", role.getId()).executeUpdate();
        // GroupProvider interface method implementation ends here
```
Apologies for not being clear before

Reply:
No apologies needed, I should have got this. Fixed